daphne/src/roles.rs

```diff
@@ -172,10 +172,21 @@ where
             }
         }
 
-        Ok(DapResponse {
-            media_type: Some(MEDIA_TYPE_HPKE_CONFIG),
-            payload: hpke_config.as_ref().get_encoded(),
-        })
+        match req.version {
+            DapVersion::Draft02 => Ok(DapResponse {
+                media_type: Some(MEDIA_TYPE_HPKE_CONFIG),
+                payload: hpke_config.as_ref().get_encoded(),
+            }),
+            _ => {
```

Comment:
This branch also matches on DapVersion::Unknown, which is not what we want to do. In general, I think it's preferable to be explicit about matching on enum variants. That way, when we add a new version number (or, more likely, replace 03 with 04), the compiler tells us exactly where we need to look in the code for changes in the protocol logic.
```suggestion
            DapVersion::Draft03 => {
```

Reply:
This didn't actually cover the Unknown case as we return with an error on DapVersion::Unknown at the top, but I'm still OK with making us explicitly update the protocol rather than have an "or other valid versions" effect with "_".

Reply:
Ah, I see. I guess my larger point is: It seems like good practice to not match on "_" when the variants can be enumerated.